source/server/listener_manager_impl.cc

```diff
@@ -688,21 +697,17 @@ void ListenerManagerImpl::stopListeners(StopListenersType stop_listeners_type) {
       // Close the socket once all workers stopped accepting its connections.
       // This allows clients to fast fail instead of waiting in the accept queue.
       const uint64_t listener_tag = listener.listenerTag();
-      stopListener(listener,
-                   [this, share_socket = listener.listenSocketFactory().sharedSocket().has_value(),
-                    listener_tag]() {
-                     stats_.listener_stopped_.inc();
-                     if (!share_socket) {
-                       // Each listener has its own socket and closes the socket
-                       // on its own.
-                       return;
-                     }
-                     for (auto& listener : active_listeners_) {
-                       if (listener->listenerTag() == listener_tag) {
-                         listener->listenSocketFactory().sharedSocket()->get().close();
-                       }
-                     }
-                   });
+      stopListener(listener, [this, listener_tag]() {
+        stats_.listener_stopped_.inc();
+        for (auto& listener : active_listeners_) {
```

Comment:
I would prefer keeping the original behavior if there is no substantial improvement because of this change. If socket is not shared, what the for loop does doesn't make such sense.

Reply:
reverted.